docs/03. Planificación/3.2. Gestión del desarrollo/Test plan.mdx

```diff
@@ -0,0 +1,220 @@
+        "date": "2025-03-08T00:00:00",
+        "reason": "Updated reason",
+        "typology": "Updated typology",
+        "observations": "Updated observations",
+        "technician": "Jane Doe",
+        "update_fields_to_none": ["reason"]
+    }
+
+    # Execute the update operation
+    response = app_client.patch(url=url, json=updated_data)
+    assert response.status_code == 200
+    result = response.json()
+
+    # Verify the updated fields
+    assert result["date"] == updated_data["date"]
+    assert result["typology"] == updated_data["typology"]
+    assert result["observations"] == updated_data["observations"]
+    assert result["technician"] == updated_data["technician"]
+
+
+def test_delete_intervention(
+        app_client: TestClient,
+        insert_interventions_mongo,
+        mongo_db: Database):
+    intervention = insert_interventions_mongo[0]
+    intervention_id = str(intervention["_id"])
+    url = f"{URL_INTERVENTION}/{intervention_id}"
+
+    response = app_client.delete(url=url)
+    assert response.status_code == 204
+
+    deleted_intervention = mongo_db["Intervention"].find_one(
+        {"_id": intervention["_id"]})
+    assert deleted_intervention is None
+```
+El código mostrado es un ejemplo concreto que nos permite comprender la estructura de la totalidad de las pruebas llevadas a cabo durante el desarrollo del proceso.
+
+El uso de pytest ofrece una forma eficaz de automatizar el proceso de prueba, lo que ahorra tiempo y esfuerzo en comparación con las pruebas manuales. Esta automatización se integra sin problemas con FastAPI, simplificando la validación del comportamiento de la API mediante la simulación de solicitudes HTTP. Además, la facilidad de configuración con fixtures facilita la creación y limpieza de datos de prueba. Las pruebas de integración cubren escenarios de uso del mundo real, detectando problemas de interacción entre componentes, como problemas de red o autenticación. Sin embargo, estas pruebas también presentan desafíos, como la dependencia de MongoDB y la complejidad en la configuración de pruebas que involucran archivos externos, lo que puede resultar en un tiempo de ejecución más largo en comparación con las pruebas unitarias. A pesar de esto, el sistema de testing proporciona un entorno controlado con fixtures para la inserción de datos, asegurando una cobertura exhaustiva del comportamiento del sistema, incluyendo la validación de respuestas HTTP. En comparación con otras pruebas, como las unitarias que se centran en funciones individuales, las pruebas de integración ofrecen una visión más amplia del sistema, aunque las pruebas de UI y de rendimiento también son cruciales para garantizar una experiencia de usuario óptima y evaluar el comportamiento bajo carga. En conclusión, el sistema de testing desempeña un papel vital en la calidad del software, adaptándose continuamente a medida que el software evoluciona para abordar nuevos desafíos y características emergentes.
```

Comment:
Instead of commenting on how important it is to implement such tests, you should mention:
- Which ones we have done
- Which ones we are going to do
- Examples of each